test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -846,6 +846,39 @@ void IndicateMissingValuesOnnxConversionTest()
             Done();
         }
 
+        [Fact]
+        void CopyColumnsOnnxTest()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var trainDataPath = GetDataPath(TestDatasets.generatedRegressionDataset.trainFilename);
+            var dataView = mlContext.Data.LoadFromTextFile<AdultData>(trainDataPath,
+                separatorChar: ';',
+                hasHeader: true);
+
+            var pipeline = mlContext.Transforms.CopyColumns("Target1", "Target");
+            var model = pipeline.Fit(dataView);
+            var transformedData = model.Transform(dataView);
+            var onnxModel = mlContext.Model.ConvertToOnnxProtobuf(model, dataView);
+
+            var onnxFileName = "copycolumns.onnx";
+            var onnxModelPath = GetOutputPath(onnxFileName);
+
+            SaveOnnxModel(onnxModel, onnxModelPath, null);
+
+            if (IsOnnxRuntimeSupported())
+            {
+                // Evaluate the saved ONNX model using the data used to train the ML.NET pipeline.
+                string[] inputNames = onnxModel.Graph.Input.Select(valueInfoProto => valueInfoProto.Name).ToArray();
+                string[] outputNames = onnxModel.Graph.Output.Select(valueInfoProto => valueInfoProto.Name).ToArray();
+                var onnxEstimator = mlContext.Transforms.ApplyOnnxModel(outputNames, inputNames, onnxModelPath);
+                var onnxTransformer = onnxEstimator.Fit(dataView);
+                var onnxResult = onnxTransformer.Transform(dataView);
+                CompareSelectedR4ScalarColumns(model.ColumnPairs[0].outputColumnName, outputNames[2], transformedData, onnxResult);
+            }
+            Done();
+        }
```

Comment:
Please write a test where input column is not present.

Reply:
A test that where the column is not present does not trigger line that wasnt covered. Neither does DropColumns. A way to trigger that code path is a contrived code path through an internal API. That line can only be triggered if you call CopyColumns and then also use the internal API to drop that column. But then that leads to a different bug with the output missing from the final model. 
I have restored the earlier behavior which is the correct behavior. 


---
In reply to: [330345862](https://github.com/dotnet/machinelearning/pull/4219#discussion_r330345862) [](ancestors = 330345862)